### PR TITLE
fix: alert box should fade away after some delay

### PR DESCRIPTION
**Pull request by anuj-kumary**
# Description
Before this error message is visible  even user is change there value or performed some other actions i have fixed this issue in this PR now alert message will disappeared after some delay.

Fixes [#72](https://github.com/Sachin-chaurasiya/Code-Snippets-Builder/issues/77)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

# Video

https://github.com/Sachin-chaurasiya/Code-Snippets-Builder/assets/65344146/8c926368-e2f4-4b99-8f1c-78eefe7c6758

### CODE REVIEW

File: src/components/Common/AuthForm/AuthForm.tsx [hunk 1]
```diff
@@ -186,7 +191,7 @@ const AuthForm: FC<AuthFormProps> = ({ formType }) => {
             </Text>
           </Box>
 
-          {formAPIError ? (
+          {showAlert ? (
```
Comment (Sachin-chaurasiya) [inline src/components/Common/AuthForm/AuthForm.tsx:194]:
```suggestion
          {showAlert && formAPIError ? (
```

File: src/components/Common/AuthForm/AuthForm.tsx [hunk 2]
```diff
@@ -49,6 +49,7 @@ const AuthForm: FC<AuthFormProps> = ({ formType }) => {
   const toast = useToast();
 
   const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
+  const [showAlert, setShowAlert] = useState<boolean>(false);
```
Comment (Sachin-chaurasiya) [inline src/components/Common/AuthForm/AuthForm.tsx:52]: Can we use useDisclosure hook here? so we don't have to maintain the state ourself.

**Review by Sachin-chaurasiya — COMMENTED** [review]
Left a few comments, thanks

File: src/components/Common/AuthForm/AuthForm.tsx (hunk 2 above)
Comment (anuj-kumary) [reply src/components/Common/AuthForm/AuthForm.tsx]: Done

File: src/components/Common/AuthForm/AuthForm.tsx (hunk 1 above)
Comment (anuj-kumary) [reply src/components/Common/AuthForm/AuthForm.tsx]: Done

**Comment by anuj-kumary** [comment]
> Left a few comments, thanks

@Sachin-chaurasiya I have addressed your comments can you please check.

**Review by Sachin-chaurasiya — APPROVED** [review]

**Pull request merged**